fluentforms/core/src/main/java/com/_4point/aem/fluentforms/impl/assembler/AssemblerServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+package com._4point.aem.fluentforms.impl.assembler;
+
+import java.util.Map;
+import java.util.Objects;
+
+import com._4point.aem.fluentforms.api.Document;
+import com._4point.aem.fluentforms.api.PathOrUrl;
+import com._4point.aem.fluentforms.api.assembler.AssemblerOptionsSetter;
+import com._4point.aem.fluentforms.api.assembler.AssemblerOptionsSpec;
+import com._4point.aem.fluentforms.api.assembler.AssemblerResult;
+import com._4point.aem.fluentforms.api.assembler.AssemblerService;
+import com._4point.aem.fluentforms.impl.UsageContext;
+import com.adobe.fd.assembler.client.PDFAConversionOptionSpec;
+import com.adobe.fd.assembler.client.PDFAConversionResult;
+import com.adobe.fd.assembler.client.PDFAValidationOptionSpec;
+import com.adobe.fd.assembler.client.PDFAValidationResult;
+
+public class AssemblerServiceImpl implements AssemblerService {
+	private final TraditionalDocAssemblerService adobDocAssemblerService;
+	private final UsageContext usageContext;
+
+	public AssemblerServiceImpl(TraditionalDocAssemblerService adobDocAssemblerService, UsageContext usageContext) {
+		super();
+		this.adobDocAssemblerService = new SafeDocAssemblerServiceAdapterWrapper(adobDocAssemblerService);
+		this.usageContext = usageContext;
+	}
+
+	@Override
+	public AssemblerResult invoke(Document ddx, Map<String, Object> sourceDocuments,
+			AssemblerOptionsSpec assemblerOptionSpec) throws AssemblerServiceException {
+		Objects.requireNonNull(ddx, "ddx Document cannot be null.");
+		Objects.requireNonNull(sourceDocuments, "sourceDocuments cannot be null.");
+		try {
```

Comment:
In general, I don't have null checks in the actual implementation class.  Instead, I have them centralized in the `Safe...AdapterWrapper` class.  This allows someone to remove the null checks if they wish (in order to speed up the code).  Your `SafeDocAssemblerServiceAdapterWrapper` class also has these checks in it, so it shouldn't be required to have then in both.  I would remove these ones here.

Reply:
Removed null check from the class